Fixes #5883 - Reflect token expiry in build status

roidelapluie opened the pull request:
The Global Status introduced in recent Foreman releases makes it easy to
mark the build status as erroneous if the token has expired. This drags
attention to the fact that the host will not be able to mark itself as
built anymore.

roidelapluie commented:
[test]

roidelapluie commented:
[test]

timogoebel commented on test/test_helper.rb:
```diff
@@ -1,10 +1,10 @@
+#require 'simplecov'
+#SimpleCov.start 'rails' do
+#  add_group 'API', 'app/controllers/api'
+#end
```
Why remove this?

roidelapluie replied:
It was a an accident

dLobatog commented on test/unit/host_test.rb:
```diff
@@ -1498,6 +1498,20 @@ def teardown
       assert_nothing_raised(ActiveRecord::InvalidForeignKey) {h.reload.destroy}
     end
 
+    test "token_expired? should be true if expiration date is in the past" do
```
Can you write an expectation instead of writing the setting to the db? Same goes for the host creation - if you just want to test `.token_expired?` you can do with just 'FactoryGirl.build_stubbed(:host, :managed)` which doesn't save anything to db. 

The 2 first lines of code in this test and the following are the same, please move them to a setup block.

dLobatog commented on test/unit/host_status/build_status_test.rb:
```diff
@@ -21,6 +21,18 @@ def setup
+      end
+
+      @status.stub(:token_expired?, false) do
+        assert_equal 'Pending installation', @status.to_label
```
These two assertions are separate tests, mind to split it in 2?

roidelapluie commented:
@dLobatog do you prefer this?

Commit:
Fixes #5883 - Reflect token expiry in build status

The Global Status introduced in recent Foreman releases makes it easy to
mark the build status as erroneous if the token has expired. This drags
attention to the fact that the host will not be able to mark itself as
built anymore.

dLobatog commented:
Merged as 5e1b787534cb4af70075640928edab5588b1c984, thanks @roidelapluie!
